### PR TITLE
#119 - Shared Folder Tests

### PR DESCRIPTION
**Pull request by Ari-S-123**
We wrote tests for all functions in pipes.test.tsx. We did not write any tests for the functions in test-utils.tsx as it was moved out of the shared folder.

### CODE REVIEW

File: src/shared/tests/pipes.test.tsx [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+describe('Formatting Dates', () => {
+  test('invalid test', () => {
+    expect(datePipe(new Date(99999999999999999999))).toBe('Invalid Date');
+  });
+  test('empty constructor/current time', () => {
+    let current_time = new Date();
+    let answer = current_time.toLocaleString('en-US');
+    expect(datePipe(new Date())).toBe(answer);
+  });
+  // This test fails on github but passes locally.
```
Comment (jamescd18) [inline src/shared/tests/pipes.test.tsx:212]: If this is still true, let's just remove it, otherwise, remove the comment

File: src/shared/tests/pipes.test.tsx [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+  });
+  test('non-empty string with numbers', () => {
+    expect(emptyStringPipe('123')).toBe('123');
+  });
+  test('non-empty string with special characters', () => {
+    expect(emptyStringPipe('!@#$%^&*()}{:">?')).toBe('!@#$%^&*()}{:">?');
+  });
+});
+
+describe('Formatting Empty Strings With Dash', () => {
```
Comment (jamescd18) [inline src/shared/tests/pipes.test.tsx:185]: Do you have any cases here that expect an em dash output?

File: src/shared/tests/pipes.test.tsx [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+  test('empty string', () => {
+    expect(emDashPipe('')).toBe('');
+  });
+  test('empty string with space', () => {
+    expect(emDashPipe(' ')).toBe(' ');
```
Comment (jamescd18) [inline src/shared/tests/pipes.test.tsx:190]: I think it'd be good to have this behave by outputting an em dash. Check out the trim method if you're not already familiar with it

File: src/shared/tests/pipes.test.tsx [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+    expect(datePipe(new Date('2021-10-5'))).toBe('10/5/2021, 12:00:00 AM');
+  });
+  test('custom detailed', () => {
+    expect(datePipe(new Date(2021, 0o11, 0o5, 14, 3, 6, 9))).toBe('10/5/2021, 2:03:06 PM');
+  });
+});
+
+describe('Formatting Wbs Numbers', () => {
+  test('with dummy data', () => {
+    expect(wbsPipe(exampleWorkPackage1.wbsNum)).toBe('1.1.1');
```
Comment (jamescd18) [inline src/shared/tests/pipes.test.tsx:226]: I think you can find example WBS#s directly in the stub data rather than pulling from projects/work packages if you want to

File: src/shared/tests/pipes.test.tsx [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+  });
+  test('datestring', () => {
+    expect(datePipe(new Date('2021-10-5'))).toBe('10/5/2021, 12:00:00 AM');
+  });
+  test('custom detailed', () => {
+    expect(datePipe(new Date(2021, 0o11, 0o5, 14, 3, 6, 9))).toBe('10/5/2021, 2:03:06 PM');
+  });
+});
+
+describe('Formatting Wbs Numbers', () => {
```
Comment (jamescd18) [inline src/shared/tests/pipes.test.tsx:224]: Let's add a test case that doesn't use the dummy data so we can have 2 and 3 digit numbers in the WBS#

File: src/shared/tests/pipes.test.tsx [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+    expect(weeksPipe(1.5)).toBe('1.5 weeks');
+  });
+  test('with hexadecimal', () => {
+    expect(weeksPipe(0x37cf)).toBe('14287 weeks');
+  });
+  test('with octal', () => {
+    expect(weeksPipe(0o377)).toBe('255 weeks');
+  });
+  test('with binary', () => {
+    expect(weeksPipe(0b111001)).toBe('57 weeks');
```
Comment (jamescd18) [inline src/shared/tests/pipes.test.tsx:131]: I love this, and it's kinda funny cuz it's valid, but this is the kinda test case that would probably be a little much on co-op. 

You can leave it in here, but typically you want to bias towards the minimal number of tests which cover a couple of each base cases and one for each major edge case and no more. This is because more tests = more work to maintain them / edit them when the application changes.
